src/lib.rs

```diff
@@ -71,18 +71,8 @@
     clippy::cast_lossless,
     // Covered by other lints
     clippy::missing_panics_doc, // clippy::unwrap_used
-    // Callbacks with `+ 'static + Send + Sync`, trigger this pretty easily
-    clippy::type_complexity,
 )]
 #![deny(rustdoc::broken_intra_doc_links, rustdoc::private_intra_doc_links)]
-#![deny(
-    clippy::unwrap_used,
-    clippy::non_ascii_literal,
-    clippy::missing_errors_doc,
-    clippy::let_underscore_must_use,
-    clippy::unused_async,
-    unused
-)]
```

Comment:
I'm not sure we need to remove these entirely. AFAIK, `unwrap_used`, `non_ascii_literal`, and `let_underscore_must_use` are part of `clippy::restriction`, so better to keep them as warnings. Plus, I'm not sure if `clippy::unused_async` is implied by Rust's `unused`, since I think that's part of the compiler itself (do they interface at all?). We need to make sure we're not accidentally allowing any of these by removing them.

Reply:
You're totally correct, thanks 😅